idx/memory/memory.go

```diff
@@ -120,6 +120,16 @@ func (m *MemoryIdx) AddOrUpdate(data *schema.MetricData, partition int32) idx.Ar
 	return archive
 }
 
+func (m *MemoryIdx) Update(entry idx.Archive) {
+	m.Lock()
+	if _, ok := m.DefById[entry.Id]; !ok {
+		m.Unlock()
+		return
+	}
+	*(m.DefById[entry.Id]) = entry
```

Comment:
This is safer. Another goroutine could already have a copy of the reference. If we just changed the address of what m.DefById[entry.Id] pointed to rather then content, then any modifications made by those other goroutines would be lost.